examples/shader-toy/main.cpp

```diff
@@ -407,13 +340,9 @@ float clickMouseY = 0.0f;
 bool firstTime = true;
 platform::TimePoint startTime;
 
-void renderFrame()
+virtual void renderFrame(int frameIndex) override
 {
-    auto frameIndex = gSwapchain->acquireNextImage();
-    if (frameIndex == -1)
-        return;
-
-    auto commandBuffer = gQueue->createCommandBuffer();
+    auto commandBuffer = gTransientHeaps[frameIndex]->createCommandBuffer();
```

Comment:
Okay, now I'm realizing that maybe I'm not understanding things. This line implies that we need N transient heaps to match an N-deep swap chain, and that the application will manually ring-buffer through the transient heaps?

My assumption had been that we'd want to have a *single* object that could be created up front with a conceptual "queue depth" and then the application could poke it to swap which if its internal frames/scopes it is on. Maybe the difference is minor, but it would be great if we can free applications from needing to always allocate these N copies of things for an N-deep swap chain.